### ISSUE TITLE
Optimum Not Working With LongT5 Summarization

### ISSUE THREAD

**Issue by reelmath** [huggingface/optimum#285]
### System Info

```shell
- `optimum` version: 1.2.3 (installed via Github installation)
- `transformers` version: 4.20.1
- Platform: Linux-5.4.188+-x86_64-with-Ubuntu-18.04-bionic
- Python version: 3.7.13
- Huggingface_hub version: 0.8.1
- PyTorch version (GPU?): 1.11.0+cu113 (False)
- Tensorflow version (GPU?): 2.8.2 (False)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: no
- Using distributed or parallel set-up in script?: no
```

### Who can help?

@lewtun @michaelbenayoun @JingyaHuang 

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

Essentially, I am unable to run LongT5 for anything longer than about ~25 tokens with the summarization pipeline. I get an error regarding a LessOrEqual node ending in "Can broadcast 0 by 0 or 1. num is invalid", where num is a value that increases the longer the sequence I provide as input to the pipeline.

If it helps, I get the same error when using Huggingface directly with !python -m transformers.onnx --model=my-custom-pretrained-summarizer --feature seq2seq-lm.

!pip install transformers
!pip install transformers[onnx]
!python -m pip install git+https://github.com/huggingface/optimum.git
!python -m pip install git+https://github.com/huggingface/optimum.git#egg=optimum[onnxruntime]
!pip install datasets

from optimum.onnxruntime import ORTModelForSeq2SeqLM

model = ORTModelForSeq2SeqLM.from_pretrained("my-custom-pretrained-summarizer", from_transformers=True)

Output:
/usr/local/lib/python3.7/dist-packages/transformers/models/longt5/modeling_longt5.py:180: TracerWarning: torch.tensor results are registered as constants in the trace. You can safely ignore this warning if you use this function to create tensors out of constant variables that would be the same every time you call this function. In any other case, this might cause the trace to be incorrect.
  _global_block_ids_lower_bound = torch.tensor(-1.0, dtype=global_block_ids.dtype, device=global_block_ids.device)
/usr/local/lib/python3.7/dist-packages/transformers/models/longt5/modeling_longt5.py:188: UserWarning: __floordiv__ is deprecated, and its behavior will change in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values. To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor').
  num_globals = seq_len // global_block_size
/usr/local/lib/python3.7/dist-packages/transformers/models/longt5/modeling_longt5.py:190: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if num_globals > 0:
/usr/local/lib/python3.7/dist-packages/transformers/models/longt5/modeling_longt5.py:217: TracerWarning: torch.tensor results are registered as constants in the trace. You can safely ignore this warning if you use this function to create tensors out of constant variables that would be the same every time you call this function. In any other case, this might cause the trace to be incorrect.
  block_ids >= 0, torch.tensor(global_seq_len, dtype=block_ids.dtype, device=block_ids.device)
/usr/local/lib/python3.7/dist-packages/transformers/models/longt5/modeling_longt5.py:217: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  block_ids >= 0, torch.tensor(global_seq_len, dtype=block_ids.dtype, device=block_ids.device)
/usr/local/lib/python3.7/dist-packages/transformers/models/longt5/modeling_longt5.py:219: TracerWarning: Converting a tensor to a Python integer might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  one_hot_block_ids = nn.functional.one_hot(block_ids.type(torch.int64), global_seq_len + 1)[:, :, :-1]
/usr/local/lib/python3.7/dist-packages/transformers/models/longt5/modeling_longt5.py:84: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if x.shape[dim] % block_len != 0:
/usr/local/lib/python3.7/dist-packages/transformers/models/longt5/modeling_longt5.py:67: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if not all(x.shape):
/usr/local/lib/python3.7/dist-packages/transformers/models/longt5/modeling_longt5.py:86: UserWarning: __floordiv__ is deprecated, and its behavior will change in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values. To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor').
  num_blocks = x.shape[dim] // block_len
/usr/local/lib/python3.7/dist-packages/transformers/models/longt5/modeling_longt5.py:89: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if 0 in output_shape:
/usr/local/lib/python3.7/dist-packages/transformers/modeling_utils.py:770: FutureWarning: The `device` argument is deprecated and will be removed in v5 of Transformers.
  "The `device` argument is deprecated and will be removed in v5 of Transformers.", FutureWarning
/usr/local/lib/python3.7/dist-packages/transformers/modeling_utils.py:781: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  if causal_mask.shape[1] < attention_mask.shape[1]:

from transformers import AutoTokenizer, pipeline

tokenizer = AutoTokenizer.from_pretrained('google/long-t5-tglobal-large')

onnx_summarization = pipeline("summarization", model=model, tokenizer=tokenizer)

text = # Something longer than ~25 tokens
pred = onnx_summarization(text)

[/usr/local/lib/python3.7/dist-packages/onnxruntime/capi/onnxruntime_inference_collection.py](https://localhost:8080/#) in run(self, output_names, input_feed, run_options)
    190             output_names = [output.name for output in self._outputs_meta]
    191         try:
--> 192             return self._sess.run(output_names, input_feed, run_options)
    193         except C.EPFail as err:
    194             if self._enable_fallback:

RuntimeException: [ONNXRuntimeError] : 6 : RUNTIME_EXCEPTION : Non-zero status code returned while running LessOrEqual node. Name:'LessOrEqual_648' Status Message: /onnxruntime_src/onnxruntime/core/providers/cpu/math/element_wise_ops.h:603 onnxruntime::Broadcaster::Broadcaster(gsl::span<const long int>, gsl::span<const long int>) largest <= 1 was false. Can broadcast 0 by 0 or 1. 16 is invalid.

### Expected behavior

[{'summary_text': '<something that should look the same or very similar to what my PyTorch model outputs'}]

**Comment by philschmid** [comment]
cc @echarlaix

**Comment by echarlaix** [comment]
Hi @reelmath, 

I was not able to reproduce the error you are describing by running the following code:
```python
from optimum.onnxruntime import ORTModelForSeq2SeqLM
from transformers import AutoTokenizer, pipeline

model_name = "google/long-t5-local-base"
model = ORTModelForSeq2SeqLM.from_pretrained(model_name, from_transformers=True)
tokenizer = AutoTokenizer.from_pretrained(model_name)
onnx_summarization = pipeline("summarization", model=model, tokenizer=tokenizer)
text = # Something longer than ~25 tokens
pred = onnx_summarization(text)
```

Can you confirm whether you are able to run it ? (to determine whether the problem comes from the specific model you are exporting)

**Comment by echarlaix** [comment]
I was able to reproduce your error with the model `google/long-t5-tglobal-base`, it looks like the issue comes from the ONNX export. When exporting the model, the default sequence length is [8](https://github.com/huggingface/transformers/blob/main/src/transformers/onnx/config.py#L73), and during the model tracing [`num_globals > 0`](https://github.com/huggingface/transformers/blob/main/src/transformers/models/longt5/modeling_longt5.py#L191) is being converted to `False`, which results in errors for inputs with sequence length higher than `global_block_size*2`. If you need a LongT5 model with transient-global attention, you can try to increase the sequence length when exporting the model (higher than `global_block_size`), but it might results in problems for short sequence length (`< global_block_size`). Because this issue concerns the ONNX export, could you open an issue on `transformers` in order to move the discussion there?
